os/dnd.h

```diff
@@ -0,0 +1,135 @@
+// LAF OS Library
+// Copyright (C) 2024  Igara Studio S.A.
+//
+// This file is released under the terms of the MIT license.
+// Read LICENSE.txt for more information.
+
+#ifndef OS_DND_H_INCLUDED
+#define OS_DND_H_INCLUDED
+#pragma once
+
+#include "base/paths.h"
+#include "base/debug.h"
+#include "gfx/point.h"
+#include "os/surface.h"
+
+#include <functional>
+#include <memory>
+
+#pragma push_macro("None")
+#undef None // Undefine the X11 None macro
+
+os::SurfaceRef default_decode_png(const uint8_t* buf, const uint32_t len);
+os::SurfaceRef default_decode_jpg(const uint8_t* buf, const uint32_t len);
+os::SurfaceRef default_decode_bmp(const uint8_t* buf, const uint32_t len);
+os::SurfaceRef default_decode_gif(const uint8_t* buf, const uint32_t len);
+
+namespace os {
+
+  class Window;
+
+#if CLIP_ENABLE_IMAGE
+  using DecoderFunc = std::function<SurfaceRef(const uint8_t* buf, const uint32_t len)>;
```

Comment:
I was about to do this yesterday, but then I saw that std::functions are used in other places in laf, so I left it as is. Now I'm going to change it, but I would like to know if the other std::functions will be replaced in the future, or there is any other reason to not use here a std::function

Reply:
Both are right. The good part of the `std::function<>` is that it supports lambda functions with captures, and the function pointer is just a function. Probably it can be a `std::function<>` if we need it (like `handleWindowResize`) anyway I'd like to find a good solution for these kind of callbacks in the API (at the moment they are just hacks).